src/jit/emitxarch.cpp

```diff
@@ -3786,7 +3790,7 @@ void emitter::emitIns_R_R_I(instruction ins, emitAttr attr, regNumber reg1, regN
     instrDesc* id = emitNewInstrSC(attr, ival);
 
     // REX prefix
-    if (IsExtendedReg(reg1, attr) || IsExtendedReg(reg2, attr))
+    if (IsExtendedReg(reg1, attr) || IsExtendedReg(reg2, attr) || TakesRexWPrefix(ins, attr))
```

Comment:
>|| TakesRexWPrefix(ins, attr) [](start = 63, length = 29)

Did you add this because of vpermq? Why is that necessary? It wouldn't work on x86.